### ISSUE TITLE
Cannot clone Tensors of enum type

### ISSUE THREAD

**Issue by XDEmer0r-L0rd-360-G0d-SlayerXD** [mratsim/Arraymancer#523]
As the name implies, there are issues with cloning some tensors.
My use case: 
```nim
import arraymancer

type
    Nest = enum
        a, b, c
    Change = Tensor[Nest]

proc layer(t: Change): Tensor[Nest] =
    return t.clone()

var ar = newTensor[Nest]([3, 3])
echo ar.layer()
```
And my simplified case:
```nim
import arraymancer

type
    Nest = enum
        a, b, c

var ar = newTensor[Nest]([3, 3])
echo ar.clone()
```

Both give an error of
```
d:\dump\Git\Hydris\temp.nim(10) temp
C:\Users\Nicholas\.nimble\pkgs\arraymancer-0.7.5\arraymancer\tensor\init_copy_cpu.nim(27) clone
C:\Users\Nicholas\.nimble\pkgs\arraymancer-0.7.5\arraymancer\tensor\higher_order_applymap.nim(85) contiguousImpl
C:\Users\Nicholas\.nimble\pkgs\arraymancer-0.7.5\arraymancer\tensor\private\p_accessors.nim(100) atIndexMut
C:\Users\Nicholas\.nimble\pkgs\arraymancer-0.7.5\arraymancer\tensor\private\p_accessors.nim(63) getIndex
C:\Users\Nicholas\.nimble\pkgs\arraymancer-0.7.5\arraymancer\tensor\private\p_checks.nim(32) check_index
Error: unhandled exception: Number of arguments: 1, is different from tensor rank: 2 [IndexDefect]
Error: execution of an external program failed: 'd:\dump\Git\Hydris\temp.exe '
```
despite a sucessfull compilation.

Am I doing something wrong, or is this a bug? I am using nim 1.4.8 with arraymancer 0.7.5

-Cheers!

**Comment by Vindaar** [comment]
Oh, this is very interesting. I'll have a look later today.

**Comment by Vindaar** [comment]
This should be fixed in `v0.7.6`, which I just tagged. Feel free to reopen, if something is still not working!
